src/react/components/common/tagInput/tagInputItem.tsx

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import { ITag } from "../../../../models/applicationState";
+import { constants } from "../../../../common/constants";
+// tslint:disable-next-line:no-var-requires
+const tagColors = require("../tagColors.json");
+
+export enum TagEditMode {
+    Color = "color",
+    Name = "name",
+}
+
+export interface ITagInputItemProps {
+    tag: ITag;
+    index: number;
+    isBeingEdited: boolean;
+    isLocked: boolean;
+    isSelected: boolean;
+    appliedToSelectedRegions: boolean;
+    onClick: (tag: ITag, ctrlKey: boolean, altKey: boolean, clickedColor?: boolean) => void;
```

Comment:
I'd recommend refactoring these params into an object to make it easier to extend and modify in the future.